src/BitmapText.cpp

```diff
@@ -122,6 +120,38 @@ BitmapText::BitmapText( const BitmapText &cpy ):
 	*this = cpy;
 }
 
+namespace {
+	// Helper functions used in DrawPrimitives.
+	inline void SetVertexColor(std::vector<RageSpriteVertex>& vertices, size_t i, const RageColor* color)
+	{
+		vertices[i + 0].c = color[0]; // top left
+		vertices[i + 1].c = color[2]; // bottom left
+		vertices[i + 2].c = color[3]; // bottom right
+		vertices[i + 3].c = color[1]; // top right
+	}
+
+	inline void ApplyJitter(std::vector<RageSpriteVertex>& vertices, size_t i, const RageVector3& jitter)
+	{
+		vertices[i + 0].p += jitter; // top left
+		vertices[i + 1].p += jitter; // bottom left
+		vertices[i + 2].p += jitter; // bottom right
+		vertices[i + 3].p += jitter; // top right
+	}
+
+	inline void UndoJitter(std::vector<RageSpriteVertex>& vertices, size_t i, const RageVector3& jitter)
+	{
+		vertices[i + 0].p -= jitter; // top left
+		vertices[i + 1].p -= jitter; // bottom left
+		vertices[i + 2].p -= jitter; // bottom right
+		vertices[i + 3].p -= jitter; // top right
+	}
+
+	inline size_t CalculateEndIndex(bool isEnd, size_t verticesSize, size_t iterFirst)
```

Comment:
Something about this function is off, while debugging in my local branch I can't get past the title screen in the debug build without this happening:

![image](https://github.com/user-attachments/assets/3a1c3f08-0911-4d4c-9ab8-b8e500cc2844)

I think it's related to casting/passing the types in? fwiw in release mode I'm not seeing this issue, and Visual Studio points me to this function and its invocations on line 783 and 860. When I revert back to ```size_t iEnd = iter == m_mAttributes.end() ? m_aVertices.size() : iter->first * 4;``` the code seems to work as expected. Weirdly enough this error is ignored/skipped in the release build

Reply:
> Something about this function is off, while debugging in my local branch I can't get past the title screen in the debug build without this happening:
> 
> ![image](https://private-user-images.githubusercontent.com/6710311/370097332-3a1c3f08-0911-4d4c-9ab8-b8e500cc2844.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjcxNDk0ODgsIm5iZiI6MTcyNzE0OTE4OCwicGF0aCI6Ii82NzEwMzExLzM3MDA5NzMzMi0zYTFjM2YwOC0wOTExLTRkNGMtOWFiOC1iOGU1MDBjYzI4NDQucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDkyNCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA5MjRUMDMzOTQ4WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZWRhNzViODc3OWJhMGU3MDczNWVhMjYzNTU0M2JjNmRlZThmZmY4NGEyMjY4YjhkYTUwNGNiYWNhOGU1ZWZkOSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.i3LlasnTddAX7xP930r8gSgtmKPIfMQffkNyoTUlPRk)
> 
> I think it's related to casting/passing the types in? fwiw in release mode I'm not seeing this issue, and Visual Studio points me to this function and its invocations on line 783 and 860. When I revert back to `size_t iEnd = iter == m_mAttributes.end() ? m_aVertices.size() : iter->first * 4;` the code seems to work as expected. Weirdly enough this error is ignored/skipped in the release build

Thank you very much for noting this. I haven't build a `debug` build in a long time  I'll look into what's the cause of this.